src/instrumentsscene/qml/MuseScore/InstrumentsScene/InstrumentsPanel.qml

```diff
@@ -166,6 +170,21 @@ Item {
                 headerVisible: false
                 frameVisible: false
 
+                function expandCollapseAll(expand) {
+                    for (let row = 0; row < instrumentsTreeView.model.rowCount(); row++) {
+                        const instrumentIndex = instrumentsTreeView.model.index(row, 0);
+                        const instrumentsTreeItemDelegate = instrumentsTreeView.model.data(instrumentIndex);
+                        if (instrumentsTreeItemDelegate.isExpandable){
+                            if (expand) {
+                                instrumentsTreeView.expand(instrumentIndex)
+                            } else {
+                                instrumentsTreeView.collapse(instrumentIndex)
+                            }
+                        }
+                    }
+                    instrumentsTreeView.flickableItem.returnToBounds();
```

Comment:
@MaxiSchwindler After some experimenting, I found that this seems to work: 
```suggestion
                    flickable.returnToBounds();
```
Why it works: as I just remembered, we've disabled the default scroll functionality from TreeView and put it inside a Flickable ourselves. So we need to return _that_ Flickable to its bounds!

It seems that the behaviour will still not be perfect when you scroll manually after using "collapse all", but that seems really a Qt bug, so that's probably the best we can get.